.github/workflows/linux-eic-shell.yml

```diff
@@ -263,52 +266,23 @@ jobs:
         path: sim_dis_${{matrix.beam}}_minQ2=${{matrix.minq2}}_${{ matrix.detector_config }}.edm4hep.root
         if-no-files-found: error
 
-  valgrind-memcheck-eicrecon-gcc:
-    runs-on: ubuntu-latest
-    needs:
-    - build
-    - ddsim-gun
-    strategy:
-      matrix:
-        particle: [pi]
-        detector_config: [arches, brycecanyon]
-    steps:
-    - uses: actions/download-artifact@v3
-      with:
-        name: install-gcc-eic-shell-Debug
-    - uses: actions/download-artifact@v3
-      with:
-        name: sim_${{ matrix.particle }}_1GeV_20GeV_${{ matrix.detector_config }}.edm4hep.root
-    - uses: cvmfs-contrib/github-action-cvmfs@v3
-    - name: Run EICrecon under valgrind
-      uses: eic/run-cvmfs-osg-eic-shell@main
-      with:
-        platform-release: "jug_xl:nightly"
-        setup: /opt/detector/setup.sh
-        run: |
-          export DETECTOR_CONFIG=${DETECTOR}_${{ matrix.detector_config }}
-          export JANA_PLUGIN_PATH=$PWD/lib/EICrecon/plugins${JANA_PLUGIN_PATH:+:${JANA_PLUGIN_PATH}}
-          chmod a+x bin/*
-          valgrind --tool=memcheck --track-origins=yes --leak-check=full --log-file=valgrind-memcheck-eicrecon-${{ matrix.particle }}-${{ matrix.detector_config }}.log --suppressions=/usr/local/etc/root/valgrind-root.supp $PWD/bin/eicrecon -Ppodio:output_file=rec_${{ matrix.particle }}_1GeV_20GeV_${{ matrix.detector_config }}.edm4eic.root sim_${{ matrix.particle }}_1GeV_20GeV_${{ matrix.detector_config }}.edm4hep.root -Pplugins=dump_flags,janadot -Pjana:timeout=0 -Pjana:warmup_timeout=0 -Pdump_flags:json=${{ matrix.detector_config }}_flags.json
-    - uses: actions/upload-artifact@v3
-      with:
-        name: valgrind-memcheck-eicrecon-${{ matrix.particle }}-${{ matrix.detector_config }}.log
-        path: valgrind-memcheck-eicrecon-${{ matrix.particle }}-${{ matrix.detector_config }}.log
-        if-no-files-found: error
-
-  eicrecon-gcc-gun:
+  eicrecon-gun:
     runs-on: ubuntu-latest
     needs:
     - build
     - ddsim-gun
     strategy:
       matrix:
+        CC: [clang]
         particle: [pi, e]
         detector_config: [arches, brycecanyon]
+    env:
+      ASAN_OPTIONS: malloc_context_size=20:detect_leaks=1:verify_asan_link_order=0:detect_stack_use_after_return=true:halt_on_error=false:detect_odr_violation=0
```

Comment:
Should we use `halt_on_error=true` in these jobs? Any other ASAN options we should enable?

Reply:
Googling suggests that it may not even work without some `-fsanitize-recover` setting on the compiler side. I think, we should just drop that option (1 is its default value).